### PR TITLE
docs(behaivor_path_planner): add documentation for new planner manager

### PR DESCRIPTION
**Pull request by satoshi-ota**
## Description

Create documentation for NEW manager.

## Tests performed

Nothing.

## Effects on system behavior

Nothing.

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: planning/behavior_path_planner/docs/behavior_path_planner_manager_design.md [hunk 1]
```diff
@@ -0,0 +1,487 @@
...
+
+Click [here](../README.md) for supported scene modules.
+
+!!! warning
+
+    It is still under development and some functions may be unstable.
+
+## Overview
+
+The manager is the implementation part of the core of `behavior_path_planner`. It outputs path based on the latest data.
```
Comment (zulfaqar-azmi-t4) [inline planning/behavior_path_planner/docs/behavior_path_planner_manager_design.md:30]:
```suggestion
The manager is the core part of `behavior_path_planner` implementation. It outputs path based on the latest data.
```

File: planning/behavior_path_planner/docs/behavior_path_planner_manager_design.md [hunk 2]
```diff
@@ -0,0 +1,487 @@
...
+
+Additionally, the manager generates root reference path, and if any other modules don't request execution, the path is used as the planning result of `behavior_path_planner`.
+
+![manager_overview](../image/manager/manager_overview.svg)
+
+### Sub-managers
+
+The sub-manager's main task is
+
+- stores the launched modules in internal vectors `registered_modules_`.
```
Comment (shmpwk) [inline planning/behavior_path_planner/docs/behavior_path_planner_manager_design.md:48]:
```suggestion
- store the launched modules in internal vectors `registered_modules_`.
```

File: planning/behavior_path_planner/docs/behavior_path_planner_manager_design.md (hunk 1 above)
Comment (satoshi-ota) [reply planning/behavior_path_planner/docs/behavior_path_planner_manager_design.md]: Fixed in https://github.com/autowarefoundation/autoware.universe/pull/3493/commits/7d79ac77e442933f2cfcd7b49ee57c030f71b189.

**Review by shmpwk — APPROVED** [review]
LGTM!

**Pull request merged**